### PR TITLE
Iter5

### PR DESCRIPTION
**Pull request by monsterr00**

### CODE REVIEW

File: cmd/agent/agent
Comment (arttet) [inline cmd/agent/agent:1]: CHANGE REQUEST: it should be removed

File: cmd/agent/main.go [hunk 1]
```diff
@@ -1,3 +1,143 @@
...
+	"os"
+	"runtime"
+	"strconv"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/monsterr00/metric-service.gittest_client/internal/config"
+)
+
+var gauge map[string]float64
```
Comment (arttet) [inline cmd/agent/main.go]: CHANGE REQUEST: please don't use global variables

File: cmd/agent/main.go [hunk 2]
```diff
@@ -1,3 +1,143 @@
...
+var counter map[string]int64
```
Comment (arttet) [inline cmd/agent/main.go]: CHANGE REQUEST: please split your application by layers

File: cmd/agent/main.go [hunk 3]
```diff
@@ -1,3 +1,143 @@
...
+	for k, v := range gauge {
+		metricGaugeURL := fmt.Sprintf("/update/gauge/%s/%f", k, v)
+		requestURL := fmt.Sprintf("%s%s%s", "http://", config.ClientOptions.Host, metricGaugeURL)
+
+		req, err := client.R().
+			SetHeader("Content-Type", "text/plain").
+			Post(requestURL)
+		if err != nil {
+			fmt.Printf("Client: error sending http-request: %s\n", err)
+			os.Exit(1)
```
Comment (arttet) [inline cmd/agent/main.go]: CHANGE REQUEST: you shouldn't panic in the middle of the program.

File: cmd/agent/main.go [hunk 4]
```diff
@@ -1,3 +1,143 @@
...
+		seed := rand.NewSource(time.Now().UnixNano())
+		random := rand.New(seed)
+		gauge["RandomValue"] = random.Float64()
+
+		counter["PollCount"] += 1
+		time.Sleep(time.Duration(config.ClientOptions.PollInterval) * time.Second)
+	}
+}
+
+func init() {
```
Comment (arttet) [inline cmd/agent/main.go]: CHANGE REQUEST: it would be better if it's in `flags.go`

File: internal/config/config.go [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package config
+
+type memStorage struct {
+	Gauge   map[string]float64
+	Counter map[string]int64
+}
+
+var MemStorage memStorage
```
Comment (arttet) [inline internal/config/config.go]: CHANGE REQUEST: please don't use global variables

File: internal/handlers/handlers.go [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/monsterr00/metric-service.gittest_client/internal/config"
+)
+
+func MainPage(res http.ResponseWriter, req *http.Request) {
+	if req.Method == http.MethodGet {
```
Comment (arttet) [inline internal/handlers/handlers.go]: CLARIFICATION: are you sure that it needs?

File: internal/handlers/handlers.go [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+		splitPath := strings.Split(req.URL.Path, "/")
+		if len(splitPath) > 3 {
+			// тип метрики
+			memType := splitPath[2]
+			// имя метрики
+			memName := splitPath[3]
+
+			switch memType {
+			case "gauge":
```
Comment (arttet) [inline internal/handlers/handlers.go]: CHANGE REQUEST: this is a magic string

File: internal/handlers/handlers.go [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+	} else {
+		http.Error(res, "Only GET requests are allowed!", http.StatusMethodNotAllowed)
+		return
+	}
+}
+
+func GetMetric(res http.ResponseWriter, req *http.Request) {
+	if req.Method == http.MethodGet {
+		splitPath := strings.Split(req.URL.Path, "/")
+		if len(splitPath) > 3 {
```
Comment (arttet) [inline internal/handlers/handlers.go]: CHANGE REQUEST: this is a magic string

**Review by arttet — CHANGES_REQUESTED** [review]
Please add
- https://about.codecov.io/blog/getting-started-with-code-coverage-for-golang/
- https://github.com/golangci/golangci-lint

File: cmd/agent/agent
Comment (monsterr00) [reply cmd/agent/agent:1]: done

File: cmd/agent/main.go (hunk 2 above)
Comment (monsterr00) [reply cmd/agent/main.go]: done

File: cmd/agent/main.go (hunk 4 above)
Comment (monsterr00) [reply cmd/agent/main.go]: done

File: internal/handlers/handlers.go (hunk 6 above)
Comment (monsterr00) [reply internal/handlers/handlers.go]: done, has removed this check

File: internal/handlers/handlers.go (hunk 8 above)
Comment (monsterr00) [reply internal/handlers/handlers.go]: done, has made local const

File: cmd/agent/main.go (hunk 1 above)
Comment (monsterr00) [reply cmd/agent/main.go]: done, has made layers

File: cmd/agent/main.go (hunk 3 above)
Comment (monsterr00) [reply cmd/agent/main.go]: done, has removed panics

File: internal/config/config.go (hunk 5 above)
Comment (monsterr00) [reply internal/config/config.go]: done, has made layers

File: internal/handlers/handlers.go (hunk 7 above)
Comment (monsterr00) [reply internal/handlers/handlers.go]: done, has made local const

File: cmd/agent/flag.go [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+			os.Exit(1)
+		}
+	}
+
+	envPollInterval, isSet := os.LookupEnv("POLL_INTERVAL")
+	if isSet {
+		config.ClientOptions.PollInterval, err = strconv.ParseInt(envPollInterval, 10, 64)
+		if err != nil {
+			fmt.Printf("wrong parametr type for POLL_INTERVAL")
+			os.Exit(1)
```
Comment (arttet) [inline cmd/agent/flag.go:38]: SUGGESTION: please don't use it

File: cmd/agent/flag.go [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/monsterr00/metric-service.gittest_client/internal/config"
+)
+
+func init() {
```
Comment (arttet) [inline cmd/agent/flag.go:12]: SUGGESTION: please don't use `init`

**Review by arttet — APPROVED** [review]

**Pull request merged**